meta-lmp-base/recipes-containers/docker/docker-ce/docker.service

```diff
@@ -34,6 +34,9 @@ StartLimitInterval=60s
 # after three consecutive failures.
 FailureAction=reboot
 
+# pre-load container images into the docker data root directory (only works for overlay2 graph driver)
+ExecStartPre=-/bin/sh -c 'mkdir -p /var/lib/docker && for f in `find /var/sota/import -name "*.tar"`; do echo "compose app: preloading $f" && tar -xf $f -C /var/lib/docker/ && rm $f; done'
+
```

Comment:
clever - so it only runs once, unless you add to that folder on subsequent boots? I'm guessing you may have thoughts on (ab)using this later on?

Reply:
I was wondering if we should just create a separated service file just for importing the containers, as that way we can probably control the process better and avoid changing the main docker service file. That way the user can add or remove the logic by deciding what to include at the lmp image.

If we create another service we can make it run before docker and only when there are known tar files at var/sota/import (e.g. `ConditionPathExistsGlob=/var/sota/import/*.tar`).

Reply:
> That way the user can add or remove the logic by deciding what to include at the lmp image.

This is actually for the container build. An assumption is that a customer makes lmp builds rarely and normally works with container build and if at some point he/she decides to get a system image pre-loaded with his/her latest apps' container images then these ci-scripts params should be turned on https://source.foundries.io/factories/msul-dev01/ci-scripts.git/tree/factory-config.yml#n38. Moreover, @doanac thinks that we should add an additional command to `fioctl` that would trigger a job (container build run) that produces a system image with pre-loaded container images taking Target as an input. Such 'interface/API' assumes that each Target's lmp build/image should include the given logic unconditionally, so users could assemble a "preloaded" image for any Target they choose.

Having said that, I think it's not a big deal to add a new a systemd service. I agree that it would "probably control the process better and avoid changing the main docker service file". I just think that we might not need it at all in the future once we move to ostree.